pipeline-service/service/src/main/java/io/harness/pms/yaml/individualschema/PipelineSchemaParserFactory.java

```
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2023 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Free Trial 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
+ */
+
+package io.harness.pms.yaml.individualschema;
+
+import io.harness.annotations.dev.HarnessTeam;
+import io.harness.annotations.dev.OwnedBy;
+
+import com.google.inject.Inject;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@OwnedBy(HarnessTeam.PIPELINE)
+public class PipelineSchemaParserFactory {
+  @Inject PipelineSchemaParserV0 pipelineSchemaParserV0;
+  private final String PIPELINE_VERSION_V0 = "v0";
+
+  public AbstractStaticSchemaParser getPipelineSchemaParser(String version) {
+    switch (version) {
+      case PIPELINE_VERSION_V0:
+        return pipelineSchemaParserV0;
+      default:
+        log.error("Pipeline schema parser not available for version: {}", version);
```

can we throw an exception maybe ?
else it will throw NPE when we do getIndividualSchema

---

done